DataFormats/Detectors/TOF/include/DataFormatsTOF/Cluster.h

```diff
@@ -0,0 +1,136 @@
+  void resetLeftContributingChannel() { mContributingChannels &= ~kLeft; }           // we do "bitwise-and" with the 26th bit only set to zero, which is the negation of "mask"
+
+  // getters of the up, down, right, left bits
+  bool isUpLeftContributing() const { return (mContributingChannels & kUpLeft) == kUpLeft ? 1 : 0; }
+  bool isUpContributing() const { return (mContributingChannels & kUp) == kUp ? 1 : 0; }
+  bool isUpRightContributing() const { return (mContributingChannels & kUpRight) == kUpRight ? 1 : 0; }
+  bool isRightContributing() const { return (mContributingChannels & kRight) == kRight ? 1 : 0; }
+  bool isDownRightContributing() const { return (mContributingChannels & kDownRight) == kDownRight ? 1 : 0; }
+  bool isDownContributing() const { return (mContributingChannels & kDown) == kDown ? 1 : 0; }
+  bool isDownLeftContributing() const { return (mContributingChannels & kDownLeft) == kDownLeft ? 1 : 0; }
+  bool isLeftContributing() const { return (mContributingChannels & kLeft) == kLeft ? 1 : 0; }
+
+  void setLabel(int label) { mLabel = label; }
+  int getLabel() const { return mLabel; }
+
+ private:
+  friend class boost::serialization::access;
+
+  float mTimeRaw;   // raw TOF time // CZ: in AliRoot it is a double
+  float mTime;      // calibrated TOF time // CZ: in AliRoot it is a double
+  float mTot;       // Time-Over-threshold // CZ: in AliRoot it is a double
+  int mL0L1Latency; // L0L1 latency // CZ: is it different per cluster? Checking one ESD file, it seems that it is always the same (see: /alice/data/2017/LHC17n/000280235/pass1/17000280235019.100/AliESDs.root)
+  int mDeltaBC;     // DeltaBC --> can it be a char or short? // CZ: is it different per cluster? Checking one ESD file, it seems that it can vary (see: /alice/data/2017/LHC17n/000280235/pass1/17000280235019.100/AliESDs.root)
+  //float  mZ;           //! z-coordinate // CZ: to be verified if it is the same in the BaseCluster class
+  float mR;                  //! radius
+  float mPhi;                //! phi coordinate
+  int mContributingChannels; // index of the channels that contributed to the cluster; to be read like this:
+                             // channel & 0x3FFFF -> first 18 bits to store the main channel
+                             // channel & bit19 (0x40000) -> alsoUPLEFT
+                             // channel & bit20 (0x80000) -> alsoUP
+                             // channel & bit21 (0x100000)-> alsoUPRIGHT
+                             // channel & bit22 (0x200000)-> alsoRIGHT
+                             // channel & bit23 (0x400000)-> alsoDOWNRIGHT
+                             // channel & bit24 (0x800000)-> alsoDOWN
+                             // channel & bit25 (0x1000000)-> alsoDOWNLEFT
+                             // channel & bit26 (0x2000000)-> alsoLEFT
+  int mLabel;                // index pointing to the first MC label in the MC label array for this cluster
```

Comment:
As far as I understand, in the clusterization you never have multiple clusters in the process of building, so, whenever you add a cluster, you can add to MCTruth all its labels. Do you need in this case the mLabel as the entry of the clusters' labels in the MCTruth? Why the cluster position in its vector cannot serve to this purpose?

Reply:
Yes, you are right, we also checked from our tests. So now "mLabel" is not there anymore.